pkg/target/kusttarget_configplugin.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package target
+
+import (
+	"github.com/pkg/errors"
+	"sigs.k8s.io/kustomize/pkg/image"
+	"sigs.k8s.io/kustomize/pkg/plugins"
+	"sigs.k8s.io/kustomize/pkg/transformers"
+	"sigs.k8s.io/kustomize/pkg/transformers/config"
+	"sigs.k8s.io/kustomize/pkg/types"
+	"sigs.k8s.io/kustomize/plugin/builtingen"
+	"sigs.k8s.io/yaml"
+)
+
+// Functions dedicated to configuring the builtin
+// transformer and generator plugins using config data
+// read from a kustomization file.
+//
+// Non-builtin plugins will get their configuration
+// from their own dedicated structs and yaml files.
+//
+// There are some loops in the functions below because
+// the kustomization file would, say, allow one to
+// request multiple secrets be made, or run multiple
+// image tag transforms, so we need to run the plugins
+// N times (plugins are easier to write, configure and
+// test if they do just one thing).
+
+func (kt *KustTarget) configureBuiltinGenerators() (
+	[]transformers.Generator, error) {
+	var result []transformers.Generator
+	for _, args := range kt.kustomization.SecretGenerator {
+		var c struct {
+			types.GeneratorOptions
+			types.SecretArgs
+		}
+		if kt.kustomization.GeneratorOptions != nil {
+			c.GeneratorOptions = *kt.kustomization.GeneratorOptions
+		}
+		c.SecretArgs = args
+		p := builtingen.NewSecretGeneratorPlugin()
+		err := kt.configureBuiltinPlugin(p, c, "secret")
+		if err != nil {
+			return nil, err
+		}
+		result = append(result, p)
+	}
+	for _, args := range kt.kustomization.ConfigMapGenerator {
+		var c struct {
+			types.GeneratorOptions
+			types.ConfigMapArgs
+		}
+		if kt.kustomization.GeneratorOptions != nil {
+			c.GeneratorOptions = *kt.kustomization.GeneratorOptions
+		}
+		c.ConfigMapArgs = args
+		p := builtingen.NewConfigMapGeneratorPlugin()
+		err := kt.configureBuiltinPlugin(p, c, "configmap")
+		if err != nil {
+			return nil, err
+		}
+		result = append(result, p)
+	}
+	return result, nil
+}
+
+// TODO:
+//
+//  1) Convert remaining legacy transformers to plugins
```

Comment:
The SMP patch transformer has k8s  dependencies. If we put it into a builtin transformer, the plugin package will have k8s dependencies.

Reply:
Yes - that problem remains.  kubectl is still a mess.